Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift

```diff
@@ -484,7 +484,9 @@ struct PlatformIntelligenceTextEffectID: Hashable {
 #if canImport(UIKit)
         UIView.animate(animation, changes: changes)
 #else
-        NSAnimationContext.animate(animation, changes: changes)
+        if #available(macOS 15, iOS 18, tvOS 18, watchOS 10, *) {
```

Comment:
This is a runtime check, which we don't need in WebKit, as we are not striving to have a single binary that runs everywhere. Furthermore,
- no need for embedded platform checks, because this is the branch where we can't import UIKit;
- no need for embedded platform checks, as these are the oldest versions supported.

As mentioned in another comment, it would be ideal to centralize HAVE-style definitions in one file, so that we have a meaningful name for the checks, and reduce the risk of using slightly different checks in different parts of the code.

Reply:
I spent some time on this, and I could not figure out how make it into a compiler check. Unfortunately, it doesn't look like `#if @available` is valid syntax, and all the valid expressions after `#if` according to the grammar don't handle this operation. My guess is that either I'm missing something, or they do it by design.

Using a definition as you suggested is probably the best way to go, I'll look into that, just didn't change it yet as I don't know this code well enough to understand everything we should compile conditionally. That will be the next step.